web-app/js/smartR/_angular/directives/conceptBox.js

```diff
@@ -76,9 +65,27 @@ window.smartRApp.directive('conceptBox', ['$rootScope', function($rootScope) {
             // this watches the childNodes of the conceptBox and updates the model on change
             new MutationObserver(function() {
                 scope.conceptGroup = _getConcepts(); // update the model
-                _setColor(); // change to red or green color if input okay
                 scope.$apply();
             }).observe(template_box, { childList: true });
+
+            scope.$watch('conceptGroup', scope.validate);
+
+            scope.validate = function() {
+                if (scope.conceptGroup.length < scope.min) {
+                    scope.instruction = 'Drag at least ' + scope.min + ' node(s) into the box';
+                    return false;
+                }
+                if (scope.conceptGroup.length > scope.max) {
+                    scope.instruction = 'Select at most ' + scope.max + ' node(s)';
+                    return false;
+                }
+                if (!_containsOnly()) {
+                    scope.instruction = 'Node(s) do not have the correct type';
+                    return false;
+                }
+                scope.instruction = '';
+                return true;
+            }
```

Comment:
I did it like that before, but changed it to this, to prevent having to validate in two different places (to show the message and to change the box styling)